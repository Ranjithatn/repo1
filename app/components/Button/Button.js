import React, { Fragment } from "react";
import Icon from "../Icon/Icon";

const Button = ({
  leftIcon,
  rightIcon,
  text,
  onClick,
  className,
  disabled = false,
  id,
  type = "button",
  style={},
}) => {
  return (
    <button
      id={id}
      type={type}
      className={"button" + (className !== undefined ? " " + className : "")}
      disabled={disabled}
      onClick={onClick}
      style={ style }
    >
      {leftIcon && <Icon icon={leftIcon} />}
      { text &&<span>{text}</span> }
      {rightIcon && <Icon icon={rightIcon} />}
    </button>
  );
};

export default Button;
