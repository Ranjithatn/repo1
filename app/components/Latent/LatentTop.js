import React, { Fragment } from "react";
import Button from "../Button/Button";

function LatentTop({ formatMessage, history }) {
  return (
    <div className="head">
      <Button
        text={formatMessage({ id: "back" })}
        className="is-primary"
        leftIcon="arrow-left"
        onClick={() => history.goBack()}
      />&nbsp;&nbsp;&nbsp;&nbsp;
      <Button
        text={formatMessage({ id: "next" })}
        className="is-primary"
        rightIcon="arrow-right"
      />&nbsp;&nbsp;&nbsp;
    </div>
  );
}

export default LatentTop;
