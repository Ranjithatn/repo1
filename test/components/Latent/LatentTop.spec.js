import React from "react";
import toJson from "enzyme-to-json";
import LatentTop from "../../../app/components/Latent/LatentTop";
import Button from "../../../app/components/Button/Button";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";


configure({ adapter: new Adapter() });

const props = {
  formatMessage: data => data.id,
  // history: goBack = () => { }
}

describe('<Latent />', () => {

  it("should render without crashing", () => {
    const component = mount(
      <LatentTop {...props} />
    );
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  test('it should call history.goBack', () => {
    const wrapper = mount(<Button />);
    const goBack = jest.fn();
    wrapper.setProps({ history: { goBack } });
  });

  describe('Testing <Button /> component', () => {
    it('Check if the component Renders', () => {
      let comp = shallow(<Button />);
      expect(comp.length).toBe(1);
    });
  })
});
