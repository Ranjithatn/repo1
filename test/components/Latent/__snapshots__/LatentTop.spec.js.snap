// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Latent /> should render without crashing 1`] = `
<LatentTop
  formatMessage={[Function]}
>
  <div
    className="head"
  >
    <Button
      className="is-primary"
      leftIcon="arrow-left"
      onClick={[Function]}
      text="back"
    >
      <button
        className="button is-primary"
        disabled={false}
        onClick={[Function]}
        style={Object {}}
        type="button"
      >
        <Icon
          icon="arrow-left"
        >
          <span
            className="icon"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            title=""
          >
            <i
              className="fa fa-arrow-left"
            />
          </span>
        </Icon>
        <span>
          back
        </span>
      </button>
    </Button>
        
    <Button
      className="is-primary"
      rightIcon="arrow-right"
      text="next"
    >
      <button
        className="button is-primary"
        disabled={false}
        style={Object {}}
        type="button"
      >
        <span>
          next
        </span>
        <Icon
          icon="arrow-right"
        >
          <span
            className="icon"
            disabled={false}
            onClick={[Function]}
            style={Object {}}
            title=""
          >
            <i
              className="fa fa-arrow-right"
            />
          </span>
        </Icon>
      </button>
    </Button>
       
  </div>
</LatentTop>
`;
